### ISSUE TITLE
[OMD Scheduler/production] Twilio::REST::RestError: [HTTP 400] 21604 : Unable to create record
A 'To' phone number is required.
https://www.twilio.com/docs/errors/21604

### ISSUE THREAD

**Issue by benjaminwood** [OfficeMomsandDads/scheduler#85]
## Backtrace

    line 7 of [PROJECT_ROOT]/app/workers/send_text_message_worker.rb: perform

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/59936/faults/46273348)

**Comment by benjaminwood** [comment]
I had a feeling this error would pop up. I removed a line of code that was preventing it before:

https://github.com/OfficeMomsandDads/scheduler/pull/60/commits/a411edb4be2c7642f90ca638c1f9a9412fd488ea

We should fix upstream, so that the job is not enqueued unless a number is present.

**Comment by dinjas** [comment]
I am a little confused about some of the notification logic with respect to needs. It looks like:

- when a need is created, a group of users is notified that a new need has opened up at their office and their ids get placed in the `notified_user_ids` field of the need
- when a need is updated, a group of users is notified that a **new** need has opened up at their office and their ids get placed in the `notified_user_ids` field of a need. However, the users that were notified when the need was created are excluded from this notification

Basically, it seems like with respect to needs, a user would never be notified more than once regardless of the number of times a need is updated. Is that correct?

It does look like all associated users will be notified when a need is deleted.

**Comment by dinjas** [comment]
Wait, I misspoke.

When a need is updated, the notified users are not added to the list of notified user ids. So a user would potentially get notified of subsequent updates (as long as they were not notified of the create).

**Comment by dinjas** [comment]
@benjaminwood - I think we can remove the guard clause now and then close this out?

**Comment by benjaminwood** [comment]
Yeah, I think we need to clean up some production data, too.
